modules/social_features/social_content_block/src/ContentBuilder.php

```diff
@@ -284,4 +286,110 @@ public static function processBlockForm(array $element, FormStateInterface $form
+        if ($entity_type_id === 'group') {
+          $query->leftJoin('post__field_recipient_group', 'pfrg', "base_table.${entity_id_key} = pfrg.field_recipient_group_target_id");
+          $query->leftJoin('votingapi_vote', 'vv', "(base_table.${entity_id_key} = vv.entity_id AND vv.entity_type=:entity_type) OR (pfrg.entity_id = vv.entity_id AND vv.entity_type='post')", ['entity_type' => $entity_type_id]);
+        }
+        // Otherwise only check direct votes.
+        elseif ($entity_type_id === 'node') {
+          $query->leftJoin('votingapi_vote', 'vv', "base_table.${entity_id_key} = vv.entity_id AND vv.entity_type=:entity_type", ['entity_type' => $entity_type_id]);
+        }
+        $query->addExpression('COUNT(vv.id)', 'vote_count');
+        // This assumes all votes are likes and all likes are equal. To
+        // support downvoting or rating, the query should be altered.
+        $query
+          ->condition('vv.type', 'like')
+          ->condition('vv.timestamp', $popularity_time_start, '>')
+          ->groupBy("base_table.${entity_id_key}")
+          ->orderBy('vote_count', 'DESC');
+        break;
+
+      // Creates a join that pulls in all related entities, taking the highest
+      // update time for all related entities as last interaction time and using
+      // that as sort value.
+      case 'last_interacted':
+        if ($entity_type_id === 'group') {
+          $query->leftJoin('group_content_field_data', 'gfd',"base_table.${entity_id_key} = gfd.gid");
+          $query->leftjoin('post__field_recipient_group', 'pst', "base_table.${entity_id_key} = pst.field_recipient_group_target_id");
+          $query->leftjoin('post_field_data', 'pfd', 'pst.entity_id = pfd.id');
+          $query->leftjoin('comment_field_data', 'cfd', '(gfd.entity_id = cfd.entity_id) OR (pfd.id = cfd.entity_id)');
+          $query->leftJoin('votingapi_vote', 'vv', '(gfd.entity_id = vv.entity_id) OR (pfd.id = vv.entity_id) OR (cfd.cid = vv.entity_id)');
+          $query->leftjoin('node_field_data', 'nfd', 'gfd.entity_id = nfd.nid');
+
+          $query->addExpression('GREATEST(COALESCE(MAX(gfd.changed), 0),
+            COALESCE(MAX(vv.timestamp), 0),
+            COALESCE(MAX(cfd.changed), 0),
+            COALESCE(MAX(nfd.changed), 0),
+            COALESCE(MAX(pfd.changed), 0))', 'newest_timestamp');
+
+          $query->groupBy("base_table.${entity_id_key}");
+          $query->orderBy('newest_timestamp', 'DESC');
+        }
+        elseif ($entity_type_id === 'node') {
+          $query->leftJoin('node_field_data', 'nfd', "base_table.${entity_id_key} = nfd.nid");
+          // Comment entity.
+          $query->leftjoin('comment_field_data', 'cfd', 'nfd.nid = cfd.entity_id');
+          // Like node or comment related to node.
+          $query->leftjoin('votingapi_vote', 'vv', '(nfd.nid = vv.entity_id) OR (cfd.cid = vv.entity_id)');
```

Comment:
The join conditions should include entity type checks because IDs may overlap. So if that's not included then a vote for a group could count as a vote for a post for example.

See https://github.com/goalgorilla/open_social/pull/1759/files#diff-07da4bebe69fec9d2e6697ade1478977R336

Reply:
Fixed